config/config.go

```diff
@@ -32,6 +33,31 @@ type ServerConfig struct {
 	KubeConfigs     map[string]*api.Config     `json:"kubeConfigs"`
 }
 
+func LoadFromPath(path string) (Config, error) {
+	cf := Config{
+		Path:    path,
+		Servers: make(map[string]*ServerConfig),
+	}
+
+	content, err := os.ReadFile(path)
+	if os.IsNotExist(err) {
+		return cf, nil
```

Comment:
What happens next in this case? Will the be an error down the line if the file doesn't exist?

Reply:
If the file doesn't exist we return a valid config that only has the `Path` and an initialized map. I'll add a comment to reduce confusion about this.

Reply:
Should we return an error instead?

Reply:
So this is used pretty much everywhere in the CLI and I think I'd like to restrict the scope of the PR a bit. Since it was design this way and was working fine, would you be okay if I kept the same behavior but instead change the name to something else like:
`LoadOrNew()`, `GetOrCreate()` ? This way the name is more clear that the function will either load the config, or return an error, or return a new config?

Reply:
@maxsokolovsky I don't agree with the suggestion that this should return an error. The file not existing is a valid condition - the cli needs to be able to handle this and create the config file during this usage.

I think the code works as-is, would not recommend any changes here.

Reply:
Oh, so if it doesn't exist, then create it? Makes sense...

Reply:
Yeah that's a valid condition for the CLI as a whole, I think in this case it is expecting to always write to the file so there's no reason to create it here, though @tomleb can correct me if  I am wrong.

Reply:
Given the old name, which implies it loads the config, It's still not great because then the function has additional, unintuitive behavior. Hence Tom's suggestion to name it something else.

Reply:
Here are some examples without a config file:

```
$ ./bin/rancher server current
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xa14689]

...

$ ./bin/rancher server ls
FATA[0000] no servers are currently configured 

$ ./bin/rancher kubectl
FATA[0000] no configuration found, run `login` 
```

So not all commands writes a config, and some panics. This isn't good, but I believe should be handled separately from this PR.